lib/src/data/repository/card_list_repository.dart

```diff
@@ -1,20 +1,33 @@
 import 'dart:convert';
-import 'package:flutter/services.dart';
+
+import 'package:hearthstoneapp/src/data/datasource/remote/api_service.dart';
 import 'package:hearthstoneapp/src/data/model/hearthstone_card.dart';
 
 class CardListRepository {
-  Future<List<HearthstoneCard>> readJson(String pathToJson) async {
+  late final ApiService datasource;
+  Future<List<HearthstoneCard>> getData({String endpoint = ''}) async {
     final List<HearthstoneCard> listOfCards = [];
 
-    final String response = await rootBundle.loadString(pathToJson);
-    final List<dynamic> data = await json.decode(response) as List<dynamic>;
+    try {
+      final response = await datasource.getData(endpoint: endpoint);
 
-    data.forEach((card) {
-      final HearthstoneCard currentCard =
-          HearthstoneCard.fromJson(card as Map<String, dynamic>);
-      listOfCards.add(currentCard);
-    });
+      if (response.statusCode == 200) {
+        final List<dynamic> jsonResponse =
+            jsonDecode(response.body) as List<dynamic>;
+        jsonResponse.forEach((card) {
+          final HearthstoneCard currentCard =
+              HearthstoneCard.fromJson(card as Map<String, dynamic>);
+          listOfCards.add(currentCard);
+        });
+      }
+    } catch (e) {
+      throw Exception('there was an error retrieving data');
+    }
 
     return listOfCards;
```

Comment:
In the case of returning an empty list, does that ensure that you had an error in the response?

Reply:
I added some logic to my bloc class to send an empty state event when receiving an empty list.